### PR TITLE
feat(tangle-dapp): Add bridge input validation

### LINKED ISSUE #2363: [TASK] Handle Input Validation on Tangle Dapp bridge
### Description
Display error if the inputs provided by the users are not valid
- Amount is empty, bigger than balance, or smaller than what is required
- Address does not have the correct format

CleanShot 2024-06-06 at 16 39 53@2x

### PR DESCRIPTION
**Pull request by vutuanlinh2k2**
## Summary of changes

_Provide a detailed description of proposed changes._

- Handle Min, Max allowed for Amount Input
- Improve Error handling for the Bridge

### Proposed area of change

_Put an `x` in the boxes that apply._

- [ ] `apps/bridge-dapp`
- [ ] `apps/hubble-stats`
- [ ] `apps/stats-dapp`
- [x] `apps/tangle-dapp`
- [ ] `apps/testnet-leaderboard`
- [ ] `apps/faucet`
- [ ] `apps/zk-explorer`
- [ ] `libs/webb-ui-components`

### Reference issue to close (if applicable)

_Specify any issues that can be closed from these changes (e.g. `Closes #233`)._

- Closes #2363 

### Screen Recording

_If possible provide a screen recording of proposed change._

https://github.com/webb-tools/webb-dapp/assets/69841784/114988aa-cb0d-42f9-9433-f681199c1f60

---

### Code Checklist

_Please be sure to add .stories documentation if any additions are made to `libs/webb-ui-components`._

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by yurixander — APPROVED** [review]

File: apps/tangle-dapp/types/bridge.ts [hunk 1]
```diff
@@ -44,3 +45,8 @@ export type BridgeTokenType = {
    */
   erc20TokenContractAddress?: Partial<Record<ChainId, HexString>>;
 };
+
+export enum BridgeWalletError {
+  MismatchEvm = 'wallet-mismatch-evm',
+  MismatchSubstrate = 'wallet-mismatch-substrate',
```
Comment (AtelyPham) [inline apps/tangle-dapp/types/bridge.ts:51]: I think we need to maintain consistency for the string enum values. I suggest changing them to all upper snake-case.

`'wallet-mismatch-evm' -> 'WALLET_MISMATCH_EVM'`.

**Review by AtelyPham — APPROVED** [review]
Left a comment, otherwise LG 👍.

File: apps/tangle-dapp/types/bridge.ts (hunk 1 above)
Comment (drewstone) [inline apps/tangle-dapp/types/bridge.ts:51]:
```suggestion
  MismatchSubstrate = 'WALLET_MISMATCH_SUBSTRATE',
```

File: apps/tangle-dapp/types/bridge.ts [hunk 2]
```diff
@@ -44,3 +45,8 @@ export type BridgeTokenType = {
    */
   erc20TokenContractAddress?: Partial<Record<ChainId, HexString>>;
 };
+
+export enum BridgeWalletError {
+  MismatchEvm = 'wallet-mismatch-evm',
```
Comment (drewstone) [inline apps/tangle-dapp/types/bridge.ts:50]:
```suggestion
  MismatchEvm = 'WALLET_MISMATCH_EVM',
```

**Review by AtelyPham — APPROVED** [review]

**Pull request merged**